scripts/assert.js

```diff
@@ -0,0 +1,26 @@
+function assertEqual(a, b) {
+    if(a !== b) {
+        var error =  new Error("Assertion failed", a, b);
+        const callerStack = error.stack.split('\n')[2].trim();
+        console.error('Assertion failed:', a, b, callerStack);
+    }
+}
+
+function assertTrue(a) {
+    if(!a) {
+        var error = new Error("Assertion failed")
+        const callerStack = error.stack.split('\n')[2].trim();
+        console.error('Assertion failed:', callerStack);
+        error.message = 'Assertion failed: ' + callerStack;
+        // error.stack = error.stack.split('\n').slice(1).join('\n');
+        // error.stack = error.stack.split('\n').slice(2).join('\n');
```

Comment:
We can catch err and track them in js, but in go, there seems a lack of the outer message in the stack, for which we may look into after

Reply:
It's great to find that we can get the full stack with err.String().
so please just ignore this comment.